pkg/secrets.go

```diff
@@ -0,0 +1,114 @@
+package whisper
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	v1 "github.com/rotationalio/whisper/pkg/api/v1"
+	"github.com/rs/zerolog/log"
+)
+
+// DefaultSecretLifetime is one week after which the secret will be deleted.
+const DefaultSecretLifetime = time.Hour * 24 * 7
```

Comment:
We can probably leverage Google Secret to do the expiration; you can provide it as an `expire_time`(UTC when the Secret is scheduled to expire) or as a `ttl` (pb duration in seconds, must be more than 60).

Reply:
Awesome! This will definitely be a good way to manage things and I like the idea of doing Google Secrets only.